include/mettle/matchers/arithmetic.hpp

```diff
@@ -19,8 +19,14 @@ namespace mettle {
       ) -> bool {
         // If one of expected or actual is NaN, mag is undefined, but that's ok
         // because we'll always return false in that case, just like we should.
-        auto mag = std::max(std::abs(expected), std::abs(actual));
-        return std::abs(actual - expected) <= mag * epsilon;
+		
+		// Usually, T is a builtin scalar type supporting standard overloads
+		// std::max<T>() and std::abs<T>().  Occasionally, however, this is not
+		// true (i.e., T is a half precision float).  Koenig lookup for abs(
+		// and max() allows T to be a user defined type, with custom overloads
+		// for these functions.
+        auto mag = max(abs(expected), abs(actual));
```

Comment:
I think you'll need a *using-declaration* here so that this works with primitive types. The comment probably isn't really needed either, since this is a really common idiom...

```suggestion
        using std::max, std::abs;
        auto mag = max(abs(expected), abs(actual));
```

Reply:
I think I've convinced myself that using ADL for both `max` and `abs` here makes logical sense. If `max` isn't found using ADL, it should just fall back to `std::max` and then everything's happy. And if `your_namespace::max` doesn't work like the "max" function should, you probably don't have an arithmetic type anyway, so it's your fault for trying to use an arithmetic matcher. :)